### PR TITLE
Lighthouse CI

### PR DESCRIPTION
**Pull request by alianza**
Github action to test Vercel preview url with Lighthouse audit and report the results
Using
- wait-for-vercel-action@v1
- lighthouse-ci-action@v7

### CODE REVIEW

File: components/common/Head/Head.tsx [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
+
+    const observer = new MutationObserver(function(mutations) {
+      mutations.map(function(mutation) {
+        if (mutation.type == "attributes" && mutation.attributeName == "data-theme") {
+          setPrimaryColor(style.getPropertyValue('--primary'))
+        }
+      });
+    });
+
+    observer.observe(document.documentElement, { attributes: true })
```
Comment (egonm12) [inline components/common/Head/Head.tsx:22]: Don't forget to disconnect the observer on unmount to prevent memory leaks!

**Review by egonm12 — CHANGES_REQUESTED** [review]

File: components/common/Head/Head.tsx (hunk 1 above)
Comment (alianza) [reply components/common/Head/Head.tsx:22]: I added observer disconnect() call in hook cleanup function

File: components/common/Head/Head.tsx [hunk 2]
```diff
@@ -20,6 +20,10 @@ const Head: FC = () => {
     });
 
     observer.observe(document.documentElement, { attributes: true })
+
+    return function cleanup() {
```
Comment (egonm12) [inline components/common/Head/Head.tsx:24]: Nice! You don't have to update it but just for your information, you can also return an anonymous function like this
```
return () => observer.disconnect()
```

Nothing wrong with your approach, just a simpler solution.

**Review by egonm12 — APPROVED** [review]

File: components/common/Head/Head.tsx (hunk 2 above)
Comment (alianza) [reply components/common/Head/Head.tsx:24]: Yes I saw that. I personally thought this was a bit more descriptive. But thank for the tip!

**Pull request merged**